source/server-discovery-and-monitoring/tests/spec/find-network-error.json

```diff
@@ -0,0 +1,144 @@
+{
+  "runOn": [
+    {
+      "minServerVersion": "4.4"
+    }
+  ],
+  "database_name": "sdam-tests",
+  "collection_name": "find-network-error",
+  "data": [
+    {
+      "_id": 1
+    },
+    {
+      "_id": 2
+    }
+  ],
+  "tests": [
+    {
+      "description": "Reset server and pool after network error on find",
+      "failPoint": {
+        "configureFailPoint": "failCommand",
+        "mode": {
+          "times": 1
+        },
+        "data": {
+          "failCommands": [
+            "find"
+          ],
+          "closeConnection": true,
+          "appName": "findNetworkErrorTest"
```

Comment:
this is kind of dumb, but node didn't recognize this because the uri spec requires `appname` be all lowercase!

Reply:
I think that's a bug in the URI options spec since it's documented as `appName` externally: https://docs.mongodb.com/manual/reference/connection-string/#urioption.appName

I think it's also a bug in node since the [connection string spec says to convert the option to lowercase](https://github.com/mongodb/specifications/blob/master/source/connection-string/connection-string-spec.rst#keys): 
> Keys are strings and the character case must be normalized by lower casing the uppercase ASCII characters A through Z; other characters are left as-is.

Reply:
That may be, but there are [spec tests](https://github.com/mongodb/specifications/blob/master/source/uri-options/tests/connection-options.yml#L10) validating the URI options spec now, so I don't think we can just use a different value. The node driver does normalize to lowercase, and then retranslates to camelCase internally, except for a few cases such as `appname` which are explicitly defined as lowercase in our specs.

Reply:
I opened https://jira.mongodb.org/browse/DRIVERS-1247 to ensure drivers normalize URI options to lowercase and update the URI options spec.

Reply:
I renamed clientOptions.appName to appname.